pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -975,10 +976,124 @@ func (r *Reconciler) setPlacementStatus(
+			Message:            "Rollout is controlled by an external controller and no resource snapshot name is observed across clusters, probably rollout has not started yet",
+			ObservedGeneration: crp.Generation,
+		})
+		// As CRP status will refresh even if the spec has not changed, we reset any unused conditions
+		// to avoid confusion.
+		for i := condition.RolloutStartedCondition + 1; i < condition.TotalCondition; i++ {
+			meta.RemoveStatusCondition(&crp.Status.Conditions, string(i.ClusterResourcePlacementConditionType()))
+		}
+		return true, nil
+	}
+
+	// All bindings have the same observed resource snapshot.
+	// We only set the ObservedResourceIndex and selectedResources, as the conditions will be set with setCRPConditions.
+	// If all clusters observe the latest resource snapshot, we do not need to go through all the resource snapshots again to collect selected resources.
+	if observedResourceIndex == crp.Status.ObservedResourceIndex {
+		crp.Status.SelectedResources = selectedResourceIDs
+	} else {
+		crp.Status.ObservedResourceIndex = observedResourceIndex
+		selectedResources, err := controller.CollectResourceIdentifiersFromClusterResourceSnapshot(ctx, r.Client, crp.Name, observedResourceIndex)
+		if err != nil {
+			klog.ErrorS(err, "Failed to collect resource identifiers from clusterResourceSnapshot", "clusterResourcePlacement", klog.KObj(crp), "resourceSnapshotIndex", observedResourceIndex)
+			return false, err
+		}
+		crp.Status.SelectedResources = selectedResources
+	}
+
+	for i := range len(selected) {
+		rolloutStartedCond := meta.FindStatusCondition(allRPS[i].Conditions, string(fleetv1beta1.ResourceRolloutStartedConditionType))
+		if !condition.IsConditionStatusTrue(rolloutStartedCond, crp.Generation) &&
+			!condition.IsConditionStatusFalse(rolloutStartedCond, crp.Generation) {
+			klog.V(2).InfoS("Placement has External rollout strategy and some cluster is in RolloutStarted Unknown state, set RolloutStarted condition to True",
+				"clusterName", allRPS[i].ClusterName, "observedResourceIndex", observedResourceIndex, "clusterResourcePlacement", klog.KObj(crp))
+			crp.SetConditions(metav1.Condition{
+				Type:               string(fleetv1beta1.ClusterResourcePlacementRolloutStartedConditionType),
+				Status:             metav1.ConditionUnknown,
+				Reason:             condition.RolloutControlledByExternalControllerReason,
+				Message:            fmt.Sprintf("Rollout is controlled by an external controller and cluster %s is in RolloutStarted Unknown state", allRPS[i].ClusterName),
+				ObservedGeneration: crp.Generation,
+			})
+			// As CRP status will refresh even if the spec has not changed, we reset any unused conditions
+			// to avoid confusion.
+			for i := condition.RolloutStartedCondition + 1; i < condition.TotalCondition; i++ {
+				meta.RemoveStatusCondition(&crp.Status.Conditions, string(i.ClusterResourcePlacementConditionType()))
+			}
```

Comment:
why remove the rest of conditions in the crp.status when we are looking at a RPS[i] condition?

Reply:
No, this is saying if there's a member cluster whose RolloutStarted condition is Unknown, we set the CRP RolloutStarted condition to Unknown too and the other conditions like Overriden, Applied etc are removed.